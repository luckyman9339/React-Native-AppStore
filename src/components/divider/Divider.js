import React, { PureComponent } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';

export default class Divider extends PureComponent {

  static propTypes = {
    gutter: PropTypes.bool,
  }

  static defaultProps = {
    gutter: false,
  }

  render() {
    const {
      gutter,
    } = this.props;
    return (
      <View style={[styles.host, gutter && styles.gutter]} />
    );
  }
}

const styles = StyleSheet.create({
  host: {
    width: '100%',
    height: StyleSheet.hairlineWidth,
    backgroundColor: '#CFCFCF',
  },

  gutter: {
    marginVertical: 16,
  },
});
