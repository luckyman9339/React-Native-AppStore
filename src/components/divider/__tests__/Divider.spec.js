import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Divider from '../Divider';

test('renders correctly', () => {
  const tree = renderer.create(<Divider gutter />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('[gutter] adds style including marginVertical', () => {
  const divider = shallow(<Divider gutter />);
  expect(divider.prop('style').find(item => !!item.marginVertical)).toBeTruthy();
});
