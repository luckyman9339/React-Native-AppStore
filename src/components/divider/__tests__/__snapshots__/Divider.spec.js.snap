// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CFCFCF",
        "height": 0.5,
        "width": "100%",
      },
      Object {
        "marginVertical": 16,
      },
    ]
  }
/>
`;
